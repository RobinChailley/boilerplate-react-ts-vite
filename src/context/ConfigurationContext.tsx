import { ConfigContextValueType } from '@context/configurationContextTypes';
import getConfigContextValue from '@context/getConfigurationContextValue';
import { createContext, useContext } from 'react';

export const ConfigContext = createContext<ConfigContextValueType>(getConfigContextValue());

export const useConfig = () => useContext(ConfigContext);
