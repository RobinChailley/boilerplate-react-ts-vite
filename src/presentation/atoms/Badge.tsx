import { useConfig } from '@context/ConfigurationContext';
import React from 'react';

interface BadgeProps {
  text: string;
}

const classes = {
  container: 'atom text-sm font-semibold mr-2 px-2.5 py-0.5 rounded bg-orange-200 text-orange-900',
};

const Badge: React.FC<BadgeProps> = ({ text }) => {
  const { translatorAdapter } = useConfig();
  const t = translatorAdapter.translate();

  return <span className={classes.container}>{t(text)}</span>;
};

export default Badge;
