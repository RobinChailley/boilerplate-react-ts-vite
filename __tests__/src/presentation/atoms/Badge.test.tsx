import { ConfigContext } from '@context/ConfigurationContext';
import Badge from '@presentation/atoms/Badge';
import renderer from 'react-test-renderer';

import getConfigContextValue from '../../../mocks/context/getConfigurationContextValue';

describe('Badge', () => {
  it('should render correctly', async () => {
    let render = renderer.create(
      <ConfigContext.Provider value={getConfigContextValue()}>
        <Badge text={''} />
      </ConfigContext.Provider>,
    );

    const snapshot = render!.toJSON();
    expect(snapshot).toMatchSnapshot();
  });
});
