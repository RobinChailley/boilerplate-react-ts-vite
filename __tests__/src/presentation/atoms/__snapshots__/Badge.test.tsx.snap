// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge should render correctly 1`] = `
<span
  className="atom text-sm font-semibold mr-2 px-2.5 py-0.5 rounded bg-orange-200 text-orange-900"
/>
`;
